internal/bridge/crunchybridgecluster/crunchybridgecluster_controller_test.go

```diff
@@ -20,33 +20,961 @@ package crunchybridgecluster
+			},
+		}
+	}
+
+	t.Run("SuccessUpgrading", func(t *testing.T) {
+		cluster := testCluster()
+		cluster.Namespace = ns
+		cluster.Status.ID = upgradingClusterId
+
+		err := reconciler.handleGetClusterUpgrade(ctx, "9012", cluster)
+		assert.NilError(t, err)
+		assert.Equal(t, *cluster.Status.OngoingUpgrade[0], v1beta1.UpgradeOperation{
+			Flavor:       "resize",
+			StartingFrom: "",
+			State:        "in_progress",
+		})
+		upgradingCondition := meta.FindStatusCondition(cluster.Status.Conditions, v1beta1.ConditionUpgrading)
+		if assert.Check(t, upgradingCondition != nil) {
+			assert.Equal(t, upgradingCondition.Status, metav1.ConditionTrue)
+			assert.Equal(t, upgradingCondition.Reason, "resize")
+			assert.Check(t, cmp.Contains(upgradingCondition.Message,
+				"Performing an upgrade of type resize with a state of in_progress."))
+		}
+	})
+
+	t.Run("SuccessNotUpgrading", func(t *testing.T) {
+		cluster := testCluster()
+		cluster.Namespace = ns
+		cluster.Status.ID = notUpgradingClusterId
+
+		err := reconciler.handleGetClusterUpgrade(ctx, "9012", cluster)
+		assert.NilError(t, err)
+		assert.Equal(t, len(cluster.Status.OngoingUpgrade), 0)
+		upgradingCondition := meta.FindStatusCondition(cluster.Status.Conditions, v1beta1.ConditionUpgrading)
+		if assert.Check(t, upgradingCondition != nil) {
+			assert.Equal(t, upgradingCondition.Status, metav1.ConditionFalse)
+			assert.Equal(t, upgradingCondition.Reason, "NoUpgradesInProgress")
+			assert.Check(t, cmp.Contains(upgradingCondition.Message,
+				"No upgrades being performed"))
+		}
+	})
+
+	t.Run("UnsuccessfulGet", func(t *testing.T) {
+		cluster := testCluster()
+		cluster.Namespace = ns
+		cluster.Status.ID = notUpgradingClusterId
+
+		err := reconciler.handleGetClusterUpgrade(ctx, "bad_api_key", cluster)
+		assert.Check(t, err != nil)
+		upgradingCondition := meta.FindStatusCondition(cluster.Status.Conditions, v1beta1.ConditionUpgrading)
+		if assert.Check(t, upgradingCondition != nil) {
+			assert.Equal(t, upgradingCondition.Status, metav1.ConditionUnknown)
+			assert.Equal(t, upgradingCondition.Reason, "UnknownUpgradeState")
+			assert.Check(t, cmp.Contains(upgradingCondition.Message,
+				"Issue getting cluster upgrade from Bridge:"))
+		}
+	})
+}
+
+func TestHandleUpgrade(t *testing.T) {
+	ctx := context.Background()
+	_, tClient := setupKubernetes(t)
+	require.ParallelCapacity(t, 0)
+
+	ns := setupNamespace(t, tClient).Name
+	clusterInBridge := testClusterApiResource()
+
+	reconciler := &CrunchyBridgeClusterReconciler{
+		Client: tClient,
+		Owner:  "crunchybridgecluster-controller",
+	}
+	reconciler.NewClient = func() bridge.ClientInterface {
+		return &TestBridgeClient{
+			ApiKey:   "9012",
+			TeamId:   "5678",
+			Clusters: []*bridge.ClusterApiResource{clusterInBridge},
+		}
+	}
+
+	t.Run("UpgradePlan", func(t *testing.T) {
+		cluster := testCluster()
+		cluster.Namespace = ns
+		cluster.Status.ID = "1234"
+		cluster.Spec.Plan = "standard-16"
```

Comment:
Maybe comment that the existing plan is different? Something like...

```suggestion
		cluster.Spec.Plan = "standard-16" // Originally "standard-8"
```

Same for some of these other tests. It'd be nice to be able to see what is different from the `testCluster` without having to look into another file